internal/pkg/secret/encryption_other.go

```diff
@@ -0,0 +1,78 @@
+//go:build linux || darwin
+// +build linux darwin
+
+package secret
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/sha256"
+	"encoding/base64"
+	"os"
+	"strconv"
+
+	"github.com/confluentinc/cli/internal/pkg/errors"
+	"github.com/denisbrodbeck/machineid"
+
+	"golang.org/x/crypto/pbkdf2"
+)
+
+func DeriveEncryptionKey(salt string) ([]byte, error) {
+	machineId, err := machineid.ID()
+	if err != nil {
+		return []byte{}, err
+	}
+	userId := strconv.Itoa(os.Getuid())
+	encryptionKey := pbkdf2.Key([]byte(machineId+userId), []byte(salt), 10000, 32, sha256.New)
+	return encryptionKey, nil
+}
+
+func Encrypt(password, salt string) (string, error) {
+	encryptionKey, err := DeriveEncryptionKey(salt)
+	if err != nil {
+		return "", err
+	}
+
+	block, err := aes.NewCipher(encryptionKey)
+	if err != nil {
+		return "", err
+	}
+
+	aesgcm, err := cipher.NewGCM(block)
+	if err != nil {
+		return "", err
+	}
+
+	encryptedPassword := aesgcm.Seal(nil, encryptionKey[:12], []byte(password), nil)
```

Comment:
@ankurjoshi9 is it ok if we reuse part of the encryption key as nonce? Or do we need to store another random variable like salt in our configs?

Reply:
encryption key can not be used as nonce. You should generate it same way as salt.